maltoolbox/attackgraph/attackgraph.py

```diff
@@ -514,57 +524,70 @@ def _generate_graph(self) -> None:
 
             attack_step_nodes = []
 
-            # TODO probably part of what happens here is already done in lang_graph
-            attack_steps = self.lang_graph._get_attacks_for_asset_type(asset.type)
+            lang_graph_asset = self.lang_graph.assets[asset.type]
+            if lang_graph_asset is None:
+                raise LookupError(
+                    f'Failed to find asset with name \"{asset.type}\" in '
+                    'the language graph.'
+                )
 
-            for attack_step_name, attack_step_attribs in attack_steps.items():
+            for attack_step_name, attack_step in \
+                    lang_graph_asset.attack_steps.items():
```

Comment:
Is `attack_step_name` used? I think you use `attack_step.name` everywhere, in which case you can loop through the values only:

```suggestion
            for attack_step in lang_graph_asset.attack_steps.values():
```

Reply:
It is used for the name field and the debug log.

Reply:
we can use attack_step.name instead right?

Reply:
It adds one more variable to keep track of which is not needed at all. You have to keep one more variable in your mind to read the code. Also, it creates confusion, doesn't the step object already have the name, why carry it around separately, is it different to step.name, etc.